bfvmm/include/hve/arch/intel_x64/ept/violation.h

```diff
@@ -0,0 +1,153 @@
+//
+// Bareflank Extended APIs
+// Copyright (C) 2018 Assured Information Security, Inc.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+#ifndef EPT_VIOLATION_INTEL_X64_H
+#define EPT_VIOLATION_INTEL_X64_H
+
+#include "../base.h"
+
+// -----------------------------------------------------------------------------
+// Definitions
+// -----------------------------------------------------------------------------
+
+namespace eapis
+{
+namespace intel_x64
+{
+namespace ept
+{
+
+class hve;
+
+class EXPORT_EAPIS_HVE violation : public base
+{
+public:
+
+    struct info_t {
+        uint64_t gva;                   // In
+        uint64_t gpa;                   // In
```

Comment:
Since the handler below is reading the qualification field, I think it makes sense to stuff it into the info struct in case a downstream handler needs it.

Thoughts?